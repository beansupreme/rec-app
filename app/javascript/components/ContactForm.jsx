import React from "react"

class ContactForm extends React.Component {
  render() {
    return (
      <div id="contact-form">
        <h4>Add your info for class updates!</h4>
      </div>
    )
  }
}

export default ContactForm