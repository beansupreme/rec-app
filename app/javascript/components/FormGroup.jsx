import React from "react"

class FormGroup extends React.Component {
  render() {
    return (
      <div className="form-group">
        <label htmlFor={this.props.inputId}>{this.props.labelText}</label>
        <input id={this.props.inputId} className="form-control" 
          type={this.props.inputType} name={this.props.inputName}
          onChange={this.props.onChange} autoComplete={this.props.autoComplete}
        />
      </div>
    )
  }
}

FormGroup.defaultProps = {
  inputType: "text",
  onChange: () => {},
  autoComplete: "on"
}

export default FormGroup;