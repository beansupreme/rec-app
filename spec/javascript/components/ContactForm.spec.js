import React, {Component} from 'react';
import ReactDOM from 'react-dom';
import { shallow, render, mount } from 'enzyme';
import ContactForm from 'components/ContactForm';
import FormGroup from 'components/FormGroup';

import renderer from 'react-test-renderer';
import sinon from 'sinon'
import axios from 'axios'

describe('<ContactForm/>', () => {
  it('shows a greeting message', () => {
    const wrapper = render(<ContactForm />);
    expect(wrapper.text()).toContain('Add your info for class updates!');
  });

  it('renders as expected', () => {
    const tree = renderer.create(<ContactForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ContactForm />, div);
  });

  it('renders several FormGroups', () => {
    const wrapper = shallow(<ContactForm />);

    expect(wrapper.find(FormGroup)).toHaveLength(4)
  });

  it('sets state for address when changed', () => {
    const wrapper = mount(<ContactForm />);

    wrapper.find('#contact_name_field').simulate('change', 'Arya');


    console.log(wrapper.state.mailing_address)
  });

  it('posts the expected values to the contacts create endpoint', () => {
    sinon.spy(axios, 'post');

    const wrapper = mount(<ContactForm />);

    wrapper.find('#contact_name_field').simulate('change', 'Arya');    
    wrapper.find('#contact_email_field').simulate('change', 'noone@gmail.com');    
    wrapper.find('#contact_telephone_field').simulate('change', '');    
    wrapper.find('#contact_mailing_address_field').simulate('change', 'Hidden temple in Braavos');    

    wrapper.find('#add-contact-form').first().simulate('submit');

    expect(axios.post.calledOnce).toEqual(true);

    let expectedPostBody = {
      contact: {
        name: 'Arya',
        email: 'noone@gmail.com',
        telephone: '',
        mailing_address: 'Hidden temple in Braavos'
      }
    };

    expect(wrapper.state().email).toEqual('foo')

    let postData = axios.post.getCall(0).args;
    console.log(postData)
    expect(postData[0]).toEqual('/contacts.json');
    expect(postData[1]).toEqual(expectedPostBody);


    axios.post.restore();
  });
});