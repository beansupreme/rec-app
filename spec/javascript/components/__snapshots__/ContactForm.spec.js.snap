// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm/> renders as expected 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
    id="contact-form"
  >
    <h4
      className="card-title"
    >
      Add your info for class updates!
    </h4>
    <form
      className="new_contact"
      id="add-contact-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="contact_name_field"
        >
          Name
        </label>
        <input
          autoComplete="name"
          className="form-control"
          id="contact_name_field"
          name="contact[name]"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="contact_email_field"
        >
          Email
        </label>
        <input
          autoComplete="email"
          className="form-control"
          id="contact_email_field"
          name="contact[email]"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="contact_telephone_field"
        >
          Telephone
        </label>
        <input
          autoComplete="tel"
          className="form-control"
          id="contact_telephone_field"
          name="contact[telephone]"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="contact_mailing_address_field"
        >
          Mailing Address
        </label>
        <input
          autoComplete="address-line1"
          className="form-control"
          id="contact_mailing_address_field"
          name="contact[mailing_address]"
          onChange={[Function]}
          type="text"
        />
      </div>
      <div
        className="actions"
      >
        <input
          className="btn btn-primary"
          id="add_contact_submit"
          type="submit"
          value="Add Contact info"
        />
      </div>
    </form>
  </div>
</div>
`;
